src/com/extendedclip/papi/expansion/checkitem/CheckItemExpansion.java

```diff
@@ -25,1350 +22,1403 @@
-          } else {
-            for (Entry<String, Integer> entry : wrapper.getNbtInts().entrySet()) {
-              if (entry.getKey().contains("..")) {
-                String[] entrySplit = entry.getKey().split("\\.\\.");
-                data = "" + nbtItem.getCompound(entrySplit[0]).getInteger(entrySplit[1]);
-              } else {
-                data = "" + nbtItem.getInteger(entry.getKey());
-              }
+            if (wrapper.shouldCheckAmount()) {
+                data = multiMod ? data += "amt:" : "";
+                data += item.getAmount() + " &r";
             }
-          }
-          
-        } else if (!nbtItem.getKeys().isEmpty()) {
-          data += "nbt:";
-          for (String entry : nbtItem.getKeys()) {
-            if (nbtItem.getType(entry).equals(NBTType.NBTTagString))
-              data += "STRING:" + entry + ":" + nbtItem.getString(entry) + "|";
-            else if (nbtItem.getType(entry).equals(NBTType.NBTTagInt))
-              data += "INTEGER:" + entry + ":" + nbtItem.getInteger(entry) + "|";
-            else if (nbtItem.getType(entry).equals(NBTType.NBTTagCompound)) {
-              if (!entry.equalsIgnoreCase("display"))
-                data += "NBTTagCompound:" + entry + ":" + nbtItem.getCompound(entry) + "|";
+            if (wrapper.shouldCheckDurability()) {
+                data = multiMod ? data += "data:" : "";
+                data += item.getDurability() + " &r";
+            }
+            if (wrapper.shouldCheckCustomData() && item.hasItemMeta() && item.getItemMeta().hasCustomModelData()) {
+                data = multiMod ? data += "custommodeldata:" : "";
+                data += item.getItemMeta().getCustomModelData() + " &r";
+            }
+            if ((wrapper.shouldCheckLoreContains() || wrapper.shouldCheckLoreEquals() || wrapper.shouldCheckMultiLoreEquals())
+                    && (item.hasItemMeta() && item.getItemMeta().hasLore())) {
+                data = multiMod ? data += "lore:" : "";
+                int line = -1;
+                try {
+                    line = Integer.parseInt(wrapper.getLore());
+                } catch (Exception ignored) {
+                }
+                if (line != -1) {
+                    data += item.getItemMeta().getLore().get(line);
+                } else {
+                    for (String s : item.getItemMeta().getLore()) {
+                        if (wrapper.shouldCheckMultiLoreEquals()) {
+                            data += s + "\n";
+                        } else {
+                            data += s + "|";
+                        }
+                    }
+                    data = data.substring(0, data.length() - 1);
+                }
+                data = data + " &r";
+
+            }
+            if (wrapper.shouldCheckHasLore()) {
+                data = multiMod ? data += "haslore:" : "";
+                if (item.hasItemMeta()) {
+                    data += item.getItemMeta().hasLore() ? "yes" : "no";
```

Comment:
Should be using PlaceholderAPIPlugin boolean values `PlaceholderAPIPlugin.booleanTrue()` `PlaceholderAPIPlugin.booleanFalse()`